---
title: "project"
author: "Aphena"
date: "25/02/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Runing all bash codes in conda environment 
1. Install Anaconda (command line version)
https://docs.anaconda.com/anaconda/install/mac-os/ 
2. Create new conda environment
conda create -n python3.8 python=3.8
3. Activate new environment
conda activate python3.8
4. Install fastqc
conda install -c bioconda fastqc
5. Install multiqc
conda install -c bioconda multiqc
6. Install cutadapt
onda config --add channels defaults
conda config --add channels bioconda
conda config --add channels conda-forge
conda install cutadapt
7. Install hisat2 and related tools
conda install hisat2
conda install -c bioconda ncbi-ngs-sdk
conda install -c bioconda samtools
conda install -c bioconda igvtools
8. Install kallisto
conda install kallisto
9. Have following subfolders in the "2021-RNAproject" folder:
- data: containing all required .fastq.gz files
- results: empty folder contains all results
- doc: containing files of project notes
- src: containing all of the code written for project
- bin: containing executed programme from src

-- Have following subfolders in the "results" folder
- fastqc_multiqc_report: empty folder for fastqc&multiqc results
- cutadapt: empty folder for .fastq.gz files after cutadapt 
- cutadapt_report: empty folder for fastqc&multiqc results after cutadapt


## Setting up labels for all samples
````{R lables} 
labels = c() #create empty vectors to which labels will be added
for (i in 9:24){
  new_label = paste("EW", i, sep = "") #create strings (e.g. "EW9")
  labels = c(labels, new_label) #adding strings to vectors
}
print(labels) #print labels
````

## Setting up labels for all SC samples
````{R SC_lables} 
SC_labels = c() #create empty vectors to which labels will be added
for (i in 9:16){
  SC_new_label = paste("EW", i, sep = "") #create strings (e.g. "EW9")
  SC_labels = c(SC_labels, SC_new_label) #adding strings to vectors
}
print(SC_labels) #print labels
````

## Setting up labels for all OP samples
````{R OP_lables} 
OP_labels = c() #create empty vectors to which labels will be added
for (i in 17:24){
  OP_new_label = paste("EW", i, sep = "") #create strings (e.g. "EW9")
  OP_labels = c(OP_labels, OP_new_label) #adding strings to vectors
}
print(OP_labels) #print labels
````

## Write commands to a bash script file in which FastQC and MultiQC could be executed
1. FastQC & MultiQC codes
````{R fastqc&multiqc}
command_vector = c("#!/bin/bash") #vector to store all commands
for (i in 1:length(labels)){
  new_command = paste("fastqc ~/Desktop/2021-RNAproject/data/", labels[i], ".fastq.gz --extract -o ~/Desktop/2021-RNAproject/results/fastqc_multiqc_report/", sep="") #creating new command lines of applying fastqc for all fastq files
  command_vector = c(command_vector, new_command) #adding new command lines into vector
} 
command_vector = c(command_vector, "multiqc ~/Desktop/2021-RNAproject/results/fastqc_multiqc_report -o ~/Desktop/2021-RNAproject/results/fastqc_multiqc_report") #creating and adding new command lines of applying multiqc for all fastq files into bash script 
print(command_vector) #print all command lines
````

2. Creating new script files
````{R}
fastqc_multiqc_file = file("fastqc_multiqc_script.sh") #creating a new bash file in the current directory
writeLines(command_vector, fastqc_multiqc_file) #write entire command vector line by line to the script file
close (fastqc_multiqc_file) #close the bash file
````

## Write commands to a bash script file in which Cutadapt and QC by FastQC and MultiQC could be executed
1. Cutadapt codes
````{R cutadapt}
command_vector2 = c("#!/bin/bash") #vector to store all commands
for (i in 1:length(labels)){
  new_command2 = paste("cutadapt -a AGATCGGAAGAGCACACGTCTG ~/Desktop/2021-RNAproject/data/", labels[i], ".fastq.gz -o ~/Desktop/2021-RNAproject/results/cutadapt/cutadapt", labels[i], ".fastq.gz --minimum-length 21", sep="") #creating new command lines of cutting adaptors for all fastq files 
  command_vector2 = c(command_vector2, new_command2) #adding new command lines  into vector 
} 
print(command_vector2) #print all command lines
````

2. FastQC & MultiQC codes for quality check
````{R fastqc&multiqc}
for (i in 1:length(labels)){
  new_command2 = paste("fastqc ~/Desktop/2021-RNAproject/results/cutadapt/cutadapt", labels[i], ".fastq.gz --extract -o ~/Desktop/2021-RNAproject/results/cutadapt_report/", sep="") #creating new command line of running fastqc in all cutadapt-applied fastq files
  command_vector2 = c(command_vector2, new_command2)
} #adding new command line into vector
command_vector2 = c(command_vector2, "multiqc ~/Desktop/2021-RNAproject/results/cutadapt_report -o ~/Desktop/2021-RNAproject/results/cutadapt_report") #creating and adding new command line of running multiqc in cutadapt-applied fastqc files for quality check into bash script 
print(command_vector2) #print all command lines
````

3. Creating new script files
````{R}
cutadapt_file = file("cutadapt_script.sh") #creating a new bash file in the current directory
writeLines(command_vector2, cutadapt_file) #write entire command vector line by line to the script file
close (cutadapt_file) #close the bash file
````

## Write commands to a bash script file in which hisat2 could be executed
1. Download reference genome (genomic.gtf, genomic.fna) from
SC refer R64: https://www.ncbi.nlm.nih.gov/assembly/GCF_000146045.2
OP refer Hansenula2: https://www.ncbi.nlm.nih.gov/assembly/GCF_000187245.1
save all SC refer in "alignment_SC", the subfolder of "results" 
save all OP refer in "alignment_OP", the subfolder of "results"

2. For all SC samples, add hisat2 index to SC reference genome and generate splice junction reference
````{R SC_index}
hisat2_vector = ("#!/bin/bash") #vector to store all commands
SC_hisat2_index_command = paste("hisat2-build -p 16 ./results/alignment_SC/R64_genome.fna ./results/alignment_SC/R64_index/R64_genome.fna", "hisat2_extract_splice_sites.py ./results/alignment_SC/R64_genomic.gtf > ./results/alignment_SC/R64_splicesites.txt", sep = "\n") #create new command lines for add hisat2 index to the reference genome and generate splice junction reference
hisat2_vector = c(hisat2_vector, SC_hisat2_index_command) #add new command lines into vector
print(hisat2_vector) #print vector
````

3. Run hisat2 for SC reads alignment to reference genome
````{R SC_hisat2}
for (i in 1:length(SC_labels)){
  SC_hisat2_command = paste ("hisat2 -p 16 -k 1 -x ./results/alignment_SC/R64_index/R64_genome.fna -U ./results/cutadapt/cutadapt", SC_labels[i], ".fastq.gz -S ", SC_labels[i], ".SAM --known-splicesite-infile ./results/alignment_SC/R64_splicesites.txt --max-intronlen 2000", sep = "") #create new command lines of running hisat2 in all SC samples
  SC_move_command = paste("mv ", SC_labels[i], ".SAM ./results/alignment_SC/SC_results/", sep = "") #create new command lines of moving hisat2 alignment results (.sam files) of all SC samples to expected folder
  hisat2_vector = c(hisat2_vector, SC_hisat2_command, SC_move_command)
} #add new command lines into vector
print(hisat2_vector) #print vector
````

4. Change .sam to .bam by samtools for igv visulization for all SC samples
````{R SC_samtools}
for (i in 1:length(SC_labels)){
  SC_change_command = paste("samtools view -S -b ./results/alignment_SC/SC_results/", SC_labels[i], ".SAM > ./results/alignment_SC/SC_results/", SC_labels[i], ".bam", sep = "") #create new command lines of changing .sam files to .bam files for each SC sample
  SC_sort_command = paste("samtools sort ./results/alignment_SC/SC_results/", SC_labels[i], ".bam -o ./results/alignment_SC/SC_results/", SC_labels[i], ".sorted.bam", sep = "") #create new command lines of sorting .bam files for each SC sample
  SC_STindex_command = paste("samtools index ./results/alignment_SC/SC_results/", SC_labels[i], ".sorted.bam", sep = "") #create new command lines of indexing .bam files for each SC sample
  hisat2_vector = c(hisat2_vector, SC_change_command, SC_sort_command, SC_STindex_command) #add new command lines into vector
}
print(hisat2_vector) #print vector
````

5. For all OP samples, add hisat2 index to OP reference genome and generate splice junction reference
````{R OP_index}
OP_hisat2_index_command = paste("hisat2-build -p 16 ./results/alignment_OP/Hansenula2_genome.fna ./results/alignment_OP/Hansenula2_index/Hansenula2_genome.fna", "hisat2_extract_splice_sites.py ./results/alignment_OP/Hansenula2_genomic.gtf > ./results/alignment_OP/Hansenula2_splicesites.txt", sep = "\n") #create new command lines for add hisat2 index to the reference genome and generate splice junction reference
hisat2_vector = c(hisat2_vector, OP_hisat2_index_command) #add new command lines into vector
print(hisat2_vector) #print vector
````

6. Run hisat2 for OP reads alignment to reference genome
````{R OP_hisat2}
for (i in 1:length(OP_labels)){
  OP_hisat2_command = paste ("hisat2 -p 16 -k 1 -x ./results/alignment_OP/Hansenula2_index/Hansenula2_genome.fna -U ./results/cutadapt/cutadapt", OP_labels[i], ".fastq.gz -S ", OP_labels[i], ".SAM --known-splicesite-infile ./results/alignment_OP/Hansenula2_splicesites.txt --max-intronlen 2000", sep = "") #create new command lines of running hisat2 in all OP samples
  OP_move_command = paste("mv ", OP_labels[i], ".SAM ./results/alignment_OP/OP_results/", sep = "") #create new command lines of moving hisat2 alignment results (.sam files) of all OP samples to expected folder
  hisat2_vector = c(hisat2_vector, OP_hisat2_command, OP_move_command)
} #add new command lines into vector
print(hisat2_vector) #print vector
````

7. Change .sam to .bam by samtools for igv visulization for all OP samples
````{R OP_samtools}
for (i in 1:length(OP_labels)){
  OP_change_command = paste("samtools view -S -b ./results/alignment_OP/OP_results/", OP_labels[i], ".SAM > ./results/alignment_OP/OP_results/", OP_labels[i], ".bam", sep = "") #create new command lines of changing .sam files to .bam files for each OP sample
  OP_sort_command = paste("samtools sort ./results/alignment_OP/OP_results/", OP_labels[i], ".bam -o ./results/alignment_OP/OP_results/", OP_labels[i], ".sorted.bam", sep = "") #create new command lines of sorting .bam files for each OP sample
  OP_STindex_command = paste("samtools index ./results/alignment_OP/OP_results/", OP_labels[i], ".sorted.bam", sep = "") #create new command lines of indexing .bam files for each OP sample
  hisat2_vector = c(hisat2_vector, OP_change_command, OP_sort_command, OP_STindex_command) #add new command lines into vector
}
print(hisat2_vector) #print vector
````

8. Create new script files
````{R}
hisat2_file = file("hisat2_script.sh") #creating a new bash file in the current directory
writeLines(hisat2_vector, hisat2_file) #write entire command vector line by line to the script file
close (hisat2_file) #close the bash file
````

## Write commands to a bash script file in which hisat2 could be executed
1. Download reference transcriptome (genomic.gtf, RNA.fna) from
SC refer R64: https://www.ncbi.nlm.nih.gov/assembly/GCF_000146045.2
OP refer Hansenula2: https://www.ncbi.nlm.nih.gov/assembly/GCF_000187245.1
save all SC refer in "kallisto_SC", the subfolder of "results" 
save all OP refer in "kallisto_OP", the subfolder of "results"

2. Add kallisto index to SC reference transcriptom 
````{R SC_index}
kallisto_vector = ("#!/bin/bash") #vector to store all commands
SC_kallisto_index_command = paste("kallisto index -k 27 -i ./results/kallisto_SC/R64_transcripts.idx ./results/kallisto_SC/R64_transcriptome.fna") #create new command lines for add kallisto index to the SC reference transcriptome
kallisto_vector = c(kallisto_vector, SC_kallisto_index_command) #add new command lines into vector
print(kallisto_vector) #print vector
````

3. Run kallisto for transcript quantification in all SC samples
````{R SC_kallisto}
for (i in 1:length(SC_labels)){
  SC_kallisto_command = paste ("kallisto quant -i ./results/kallisto_SC/R64_transcripts.idx -o ./results/kallisto_SC/", SC_labels[i], "_results -b 100 --single -l 180 -s 20 --genomebam --gtf ./results/kallisto_SC/R64_genomic.gtf ./results/cutadapt/cutadapt", SC_labels[i], ".fastq.gz", sep = "") #create new command lines of running kallisto in all SC samples and generating .bam files for visulisation
  kallisto_vector = c(kallisto_vector, SC_kallisto_command)
} #add new command lines into vector
print(kallisto_vector) #print vector
````

4. Add kallisto index to OP reference transcriptom 
````{R OP_index}
OP_kallisto_index_command = paste("kallisto index -k 27 -i ./results/kallisto_OP/Hansenula2_transcripts.idx ./results/kallisto_OP/Hansenula2_transcriptom.fna") #create new command lines for add kallisto index to the OP reference transcriptome
kallisto_vector = c(kallisto_vector, OP_kallisto_index_command) #add new command lines into vector
print(kallisto_vector) #print vector
````

5. Run kallisto for transcript quantification in all SC samples
````{R OP_kallisto}
for (i in 1:length(OP_labels)){
  OP_kallisto_command = paste ("kallisto quant -i ./results/kallisto_OP/Hansenula2_transcripts.idx -o ./results/kallisto_OP/", OP_labels[i], "_results -b 100 --single -l 180 -s 20 --genomebam --gtf ./results/kallisto_OP/Hansenula2_genomic.gtf ./results/cutadapt/cutadapt", OP_labels[i], ".fastq.gz", sep = "") #create new command lines of running kallisto in all OP samples and generating .bam files for visulisation
  kallisto_vector = c(kallisto_vector, OP_kallisto_command)
} #add new command lines into vector
print(kallisto_vector) #print vector
````

6. Create new script files
````{R}
kallisto_file = file("kallisto_script.sh") #creating a new bash file in the current directory
writeLines(kallisto_vector, hisat2_file) #write entire command vector line by line to the script file
close (kallisto_file) #close the bash file
````